### PR TITLE
Pivotal ID # 184033688: Async Validation Process

### PR DESCRIPTION
**Pull request by jhoanmanuelms**
https://www.pivotaltracker.com/story/show/184033688

- Set the request processing field as nullable
- Add the draft related status
- Create the SubmissionRequestDraftService which will replace the SubmissionDraftService when drafts and requests are merged

### CODE REVIEW

File: commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/RequestStatus.kt [hunk 1]
```diff
@@ -1,6 +1,9 @@
 package ebi.ac.uk.model
 
 enum class RequestStatus {
+    DRAFT,
+    SUBMITTED,
+    ACCEPTED,
```
Comment (Juan-EBI) [inline commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/RequestStatus.kt:6]: a draft accepted is the same as request requrested is not it?

File: submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestCleanIndexer.kt [hunk 2]
```diff
@@ -43,7 +43,7 @@ class SubmissionRequestCleanIndexer(
         processId: String,
     ): SubmissionRequest =
         requestService.onRequest(accNo, version, RequestStatus.LOADED, processId) {
-            val indexedRequest = indexRequest(it.process.submission)
+            val indexedRequest = indexRequest(it.process!!.submission)
```
Comment (Juan-EBI) [inline submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestCleanIndexer.kt:46]: there is something missing in the modeling IMO that we need to use so much the it.process!!, I am afraid I can not see it now but we can take a look and improve after the completion of this feature

**Review by Juan-EBI — COMMENTED** [review]
LGTM please check minor comment

File: submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestCleanIndexer.kt (hunk 2 above)
Comment (jhoanmanuelms) [reply submission/submission-core/src/main/kotlin/ac/uk/ebi/biostd/submission/domain/request/SubmissionRequestCleanIndexer.kt:46]: I completely agree. Ideally we should be able to have a class with and without without too much mapping

File: commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/RequestStatus.kt (hunk 1 above)
Comment (jhoanmanuelms) [reply commons/commons-bio/src/main/kotlin/ebi/ac/uk/model/RequestStatus.kt:6]: Yes, we don't need this stage anymore

**Review by Juan-EBI — APPROVED** [review]
LGTM

**Pull request merged**